src/H5Tpkg.h

```diff
@@ -136,8 +144,6 @@
  * floating exception. */
 #if (H5_WANT_DATA_ACCURACY && defined(H5_LDOUBLE_TO_LLONG_ACCURATE)) || (!H5_WANT_DATA_ACCURACY)
 #define H5T_CONV_INTERNAL_LDOUBLE_ULLONG 1
-#else
-#define H5T_CONV_INTERNAL_LDOUBLE_ULLONG 0
```

Comment:
The related code that was checking this macro used `#ifdef`, not `#if`, so change this to just not define the macro rather than defining it to 0.